contents/tutorials/compare-aws-bedrock-prompts.md

```diff
@@ -0,0 +1,331 @@
+        generation_token_count: generation_token_count,
+        response_time_in_ms: responseTime
+      }
+    });
+
+    // rest of your code
+  }
+```
+
+Then, to track errors, we capture `bedrock_error` events in the `catch` block of our code:
+
+```js file=src/app/api/generate-llm-output/route.js
+// your existing code
+  } catch (error) {
+    console.error('Error:', error);
+
+    posthog.capture({
+      distinctId: email,
+      event: 'bedrock_error',
+      properties: {
+        promptId, // unique identifier for the prompt
+        prompt,
+        error_message: error.message,
+        error_name: error.name,
+        error_code: error.$metadata?.httpStatusCode,
+        error_type: error.__type,
+        error_requestId: error.$metadata?.requestId,
+        error_stack: error.stack,
+        model_id: modelId,
+      }
+    });
+
+    return new Response(JSON.stringify({ error: 'Failed to generate prompt' }), {
+      status: 500,
+      headers: { 'Content-Type': 'application/json' },
+    });
+  }
+```
+
+Refresh your app and submit a few prompts. You should then see your events captured in the [PostHog activity tab](https://us.posthog.com/events).
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1723020434/posthog.com/contents/Screenshot_2024-08-07_at_9.46.52_AM.png" 
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1723020434/posthog.com/contents/Screenshot_2024-08-07_at_9.47.04_AM.png" 
+  alt="AWS Bedrock events in PostHog" 
+  classes="rounded"
+/>
+
+#### How to create insights
+
+Now that we're capturing events, we can create [insights](/docs/product-analytics/insights) in PostHog to visualize our data. Most importantly, we breakdown our data by `promptId` to compare them.
+
+Below is an example of how to create an insight to compare average API response time for each prompt:
+
+1. Go to the [Product analytics tab](https://us.posthog.com/insights) in PostHog and click **+ New insight**.
+1. Set the event to `bedrock_completion`
+2. Click on **Total count** to show a dropdown. Click on **Property value (average)**.
+3. Select the `response_time_in_ms` property.
+4. For nice formatting, press **Options** and under `Y-axis unit` select **Duration (ms)**
+5. Click **+ Add breakdown** and select `promptId`. 
+6. **Save** your insight.
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1724925781/posthog.com/contents/Screenshot_2024-08-29_at_12.02.34_PM.png" 
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1724925781/posthog.com/contents/Screenshot_2024-08-29_at_12.02.49_PM.png" 
+  alt="Response time by prompt in PostHog" 
+  classes="rounded"
+/>
+
+### User feedback
+
+A good way to evaluate LLM outputs is to ask your users to rate them. In our sample app, we do this by asking if the response was helpful. Users can submit their response using the **Yes** and **No** buttons at the bottom of the page. 
```

Comment:
This would be a nice use case for event-based surveys (maybe for future tutorials if you are planning them)